src/standalone/Bucket/Bucket.jl

```diff
@@ -553,6 +553,31 @@ function ClimaLand.get_drivers(model::BucketModel)
     return (model.atmos, model.radiation)
 end
 
+"""
+    ClimaLand.surface_air_density(
+                    atmos::CoupledAtmosphere,
+                    model::AbstractModel,
+                    Y,
+                    p,
+                    _...,
+                )
+Returns the air density at the surface in the case of a coupled simulation.
+
+This requires the field `ρ_sfc` to be present in the cache `p` under the name
+of the model.
+"""
+function surface_air_density(
+    atmos::CoupledAtmosphere,
+    model::BucketModel,
+    Y,
+    p,
+    _...,
+)
```

Comment:
This was in `src/shared_utilities/drivers.jl` before, but for the full land model we dispatch on `NamedTuple` instead of `CoupledAtmosphere`. This means that this function only applies to the bucket, so I moved it here

Reply:
Ooo we might need to be careful here - and remove this method? I am looking here: https://github.com/CliMA/ClimaLand.jl/blob/1176a957dfd72ed69e809b8e8501dbabd52c9a1b/src/standalone/Bucket/Bucket.jl#L409C9-L409C12

and wondering if this now becomes circular (doing nothing)? Maybe we want the default method. But maybe this is OK because the coupler has already updated p.bucket.rho_sfc, which is why this does nothing.

Reply:
The last sentence is right - the coupler will update air density [here](https://github.com/CliMA/ClimaCoupler.jl/blob/eb8d2ddd589bea906484683e843a8c8e34ea1af5/src/FieldExchanger.jl#L171), so we don't need to compute it at all for a coupled bucket.

Actually now that I look at it again, the air density of the integrated land model will also get updated in the coupler at each timestep, so maybe this method does actually work for the coupled full model as well.

Reply:
Right now, the coupler computes \rho and updates it for the bucket, but not for the integrated land model. We can unify these two by adding the atmosphere thermal state to `CoupledAtmosphere` in the bucket case, and computing `\rho` within ClimaLand for all cases. Kat and I will work on this in a separate PR.